modules/widgets/social-icons.php

```diff
@@ -0,0 +1,500 @@
+<?php
+class Jetpack_Widget_Social_Icons extends WP_Widget {
+	/**
+	 * @var array Default widget options.
+	 */
+	protected $defaults;
+
+	/**
+	 * @var array protocols that are allowed in esc_url validation function.
+	 */
+	protected $protocols = array( 'http', 'https', 'mailto', 'skype' );
+
+	/**
+	 * Widget constructor.
+	 */
+	public function __construct() {
+		$widget_ops = array(
+			'classname'                   => 'jetpack_widget_social_icons',
+			'description'                 => esc_html__( 'Add social-media icons to your site.', 'jetpack' ),
```

Comment:
Could we be escaping too early here?  Would this string get escaped when actually used?

I'm concerned both with double escaping, and wrong-context escaping, if it's later used in an attribute and then gets re-escaped for that.